admin/google_search_console/class-gwt-category-filters.php

```diff
@@ -0,0 +1,156 @@
+		}
+
+		$this->set_filter_values();
+	}
+
+	/**
+	 * Getting the current view
+	 */
+	public function current_view() {
+		$view        = ( $status = filter_input( INPUT_GET, 'category' )) ? $status : 'not_found';
+		$mapped_view = WPSEO_GWT_Mapper::category( $view );
+
+		if ( filter_input( INPUT_GET, 'category' ) === null && empty( $this->category_counts[ $mapped_view ] ) ) {
+			$view = WPSEO_GWT_Mapper::category( key( $this->category_counts ), true );
+		}
+
+		return $this->current_view = $view;
+	}
+
+	/**
+	 * Returns the current filters as an array
+	 *
+	 * Only return categories with more than 0 issues
+	 *
+	 * @return array
+	 */
+	public function as_array() {
+		$new_views = array();
+
+		foreach ( $this->category_counts as $category_name => $category ) {
+			$new_views[] = $this->create_view_link( WPSEO_GWT_Mapper::category( $category_name, true ), $category['count'] );
+		}
+
+		return $new_views;
+	}
+
+	/**
+	 * Getting the options with the counts
+	 *
+	 * @return mixed
+	 */
+	private function get_counts() {
+		return get_option( WPSEO_Crawl_Issue_Count::OPTION_CI_COUNTS, array() );
+	}
+
+	/**
+	 * Setting the values for the filter
+	 */
+	private function set_filter_values() {
+		$this->set_filter_value( 'access_denied', __( 'Access denied', 'wordpress-seo' ), __( 'Server requires authentication or is blocking Googlebot from accessing the site.', 'wordpress-seo' ) );
+		$this->set_filter_value( 'faulty_redirects', __( 'Faulty redirects', 'wordpress-seo' ) );
+		$this->set_filter_value( 'not_followed',__( 'Not followed', 'wordpress-seo' ) );
+		$this->set_filter_value( 'not_found', __( 'Not found', 'wordpress-seo' ), __( 'URL points to a non-existent page.', 'wordpress-seo' ) );
+		$this->set_filter_value( 'other', __( 'Other', 'wordpress-seo' ), __( 'Google was unable to crawl this URL due to an undetermined issue.', 'wordpress-seo' ) );
+		$this->set_filter_value( 'roboted', __( 'Blocked', 'wordpress-seo' ), __( 'Googlebot could access your site, but certain URLs are blocked for Googlebot-mobile for smartphones in your robots.txt file.', 'wordpress-seo' ) );
+		$this->set_filter_value( 'server_error', __( 'Server Error', 'wordpress-seo' ), __( 'Request timed out or site is blocking Google.', 'wordpress_seo' ) );
+		$this->set_filter_value( 'soft_404', __( 'Soft 404', 'wordpress-seo' ), __( "The target URL doesn't exist, but your server is not returning a 404 (file not found) error.", 'wordpress-seo' ) );
+	}
+
+	/**
+	 * Add new filter value to the filter_values
+	 * @param string $key
+	 * @param string $value
+	 * @param string $description
+	 */
+	private function set_filter_value( $key, $value, $description = '' ) {
+		$this->filter_values[ $key ] = array(
+			'value'       => $value,
+			'description' => $description,
+		);
+	}
+
+	/**
+	 * Creates a filter link
+	 *
+	 * @param string  $key
+	 * @param integer $count
+	 *
+	 * @return string
+	 */
+	private function create_view_link( $key, $count ) {
```

Comment:
This function is not very readable.

In this instance I would prefer using `if(){}else{}` instead of `?:`. Also I would prefer using sprintf on the last line.